extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -63,28 +62,299 @@ import {
 } from "popup/ducks/transactionSubmission";
 
 import { Account } from "@shared/api/types";
+import { FlaggedKeys } from "types/transactions";
 import { AppDispatch } from "popup/App";
 
 import "./styles.scss";
+import { TransactionInfo } from "popup/components/signTransaction/TransactionInfo";
+import { BlobToSign } from "helpers/urls";
 
 export const SignTransaction = () => {
-  const { t } = useTranslation();
   const location = useLocation();
+  const blobOrTx = getTransactionInfo(location.search);
+
+  if ("blob" in blobOrTx) {
+    return (
+      <SignBlobBody blob={blobOrTx} />
+    )
+  }
+
+  return (
+    <SignTxBody
+      tx={blobOrTx}
+     />
+  )
+};
+
+interface SignBlobBodyProps {
+  blob: BlobToSign
+}
+
+const SignBlobBody = ({ blob }: SignBlobBodyProps) => {
```

Comment:
what are the differences between this and SignTxBody? It seems like there's some repeated logic here. Is it possible to either combine these into one function or split out the reusable stuff? (for ex: selecting the correct account, checking password, showing warning messages, etc.)

Reply:
yeah I think so, looking at it more I think we can probably define something like `isBlob` because the main difference is you don't show some elements for blobs and change some copy slightly.

Reply:
I refactored this in cddc615c530f802fc23ca3652c2fbb9a1ccc042b to move common state/dom up to the parent, there is still some repetition in the markup but it seemed better to me to have this bit of repetition for now vs trying to make one common render with more complicated logic to switch between blob or tx mode, I suspect that the render will only get more complicated as we expand on how we display common blobs. What do you think of this approach? @piyalbasu